app/code/Magento/CatalogGraphQl/Model/ProductOption/ProductOption.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\ProductOption;
+
+/**
+ * @inheritDoc
+ */
+class ProductOption implements ProductOptionInterface
+{
+    /**
+     * @var int|null
+     */
+    private $id;
+
+    /**
+     * @var string|null
+     */
+    private $source;
+
+    /**
+     * @var string|null
+     */
+    private $type;
+
+    /**
+     * @var string|null
+     */
+    private $value;
+
+    /**
+     * @inheritDoc
+     */
+    public function setSource(string $source): ProductOptionInterface
```

Comment:
Now we have a mechanism for using DTO without setters
See example https://github.com/magento/async-import/blob/2.3-asynchronous-import-module/app/code/Magento/AsynchronousImportSourceDataRetrieving/Model/Source.php